test/e2e/test/autoops/builder.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package autoops
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	autoopsv1alpha1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/autoops/v1alpha1"
+	controllerautoops "github.com/elastic/cloud-on-k8s/v3/pkg/controller/autoops"
+	"github.com/elastic/cloud-on-k8s/v3/test/e2e/cmd/run"
+	"github.com/elastic/cloud-on-k8s/v3/test/e2e/test"
+)
+
+type Builder struct {
+	AutoOpsAgentPolicy autoopsv1alpha1.AutoOpsAgentPolicy
+	ConfigSecret       corev1.Secret
+
+	Suffix string
+}
+
+var _ test.Builder = (*Builder)(nil)
+
+func (b Builder) SkipTest() bool {
+	return false
+}
+
+func NewBuilder(name string) Builder {
+	suffix := rand.String(4)
+	// This will place the policy in a separate namespace from the Elasticsearch instance
+	namespace := test.Ctx().ManagedNamespace(1)
```

Comment:
Why not follow the established pattern. You seem to set the namespace in the test to a different one anyway. I think that latter is much more transparent to readers of the test code.